### DISCUSSION TITLE
How to use Webapi.Performance.now()

### DISCUSSION THREAD

**Discussion by t-gebauer** [TheSpyder/rescript-webapi#104] [Q&A] [answered]
[Webapi.Performance](https://github.com/tinymce/rescript-webapi/blob/main/src/Webapi/Webapi__Performance.res) is defined as:

```rescript
type t

@send external now: t => float = "now"
```

`now()` requires an argument of type `t`, but I could not find a way to get a value of type `t` with this library.

I ended up writing my own `external` for it.

```rescript
  @val external performance: Webapi.Performance.t = "performance"
```

Shouldn't this be equivalent to how `document` and `window` [are handled](https://github.com/tinymce/rescript-webapi/blob/2a3dd9ecd0960e56c2763fa8b6530ee67cb6206e/src/Webapi/Webapi__Dom.res#L86)?

```rescript
@val external window: Dom.window = "window"
@val external document: Dom.document = "document"
```

**Comment by spocke** [comment, marked as answer]
It's defined here it takes a window instance and returns a performance t instance.

https://github.com/tinymce/rescript-webapi/blob/main/src/Webapi/Dom/Webapi__Dom__Window.res#L54

So `window->Window.performance->Performance.now` should work.

**Comment by TheSpyder** [comment]
The thing to remember is we don't use implicit globals. The only exception we make is `document`, because it's so common, everything else must come via `window`.
https://github.com/tinymce/rescript-webapi/blob/9f616918907ea67b29a4174fd851be0caea8e617/src/Webapi/Webapi__Dom.res#L86-L89

This is done both to better match the spec and also handle situations where a reference from a different window (such as an iframe) needs to be used.
